mica/report/report.py

```diff
@@ -512,7 +512,7 @@ def main(obsid):
         if summary is None:
             raise ValueError("Obsid not found in target table")
         report_status['ocat'] = summary['status']
-        links = obs_links(obsid, summary['seq_nbr'], summary['lts_lt_plan'])
+        links = obs_links(obsid, summary['seq_nbr'], str(summary['lts_lt_plan']))
```

Comment:
Unicode sandwich?  Is Ska.DBI.DBI returning `bytes` for a Sybase query?  I'm a little surprised, but if so we should fix that upstream in `Ska.DBI`.

Reply:
Not bytes.  Ska.DBI is returning a datetime.datetime (I think) and the downstream handling is a little different.

Reply:
Astropy.Time accepts `datetime` , so maybe with your new code it just works.  Bigger point though is to include comments on any code that is not obvious, like why is something being cast as `str` (esp. in this context when it used to work without the cast).

Reply:
I suppose I should just comment up the PR more for that?  It seems odd to have a comment in the code like "the python 2 interface to sql was returning something different here but now that it is a datetime it seems less trouble to just stringify it when passing to a routine that previously was just getting a string".

Reply:
An appropriate comment might be
```
# Note that summary['lts_lt_plan'] is a datetime but `obs_links()` requires a str
```
From the original code I'd say that this `lts_lt_plan` was always a `datetime` (from the attributes that get used).  I'm not even sure why it would fail now without the `str()` wrapper.

Reply:
Yeah I was planning to go with your "maybe it just works" and shouldn't have fought over the WIP implementation when I was probably going to come back these.  The only real issue with datetime I seem to have hit was that the py3 datetime.datetime was missing dayofyear .

Reply:
FYI it's a little obscure, but day of year is there as yday in the `timetuple()` output: https://docs.python.org/3/library/datetime.html#datetime.date.timetuple.

Reply:
Thanks!  But the code was using dayofyear to build YYYY:DOY:HH:MM:SS.S , and I think that is now unnecessary.